libmamba/include/mamba/specs/platform.hpp

```diff
@@ -33,12 +40,24 @@ namespace mamba::specs
         win_32,
         win_64,
         win_arm64,
+
+        // For reflexion purposes only
+        count_,
     };
 
+    constexpr auto known_platforms_count() -> std::size_t
+    {
+        return static_cast<std::size_t>(Platform::count_);
+    }
+
+    constexpr auto known_platforms() -> std::array<Platform, known_platforms_count()>;
```

Comment:
There is already the notion of `known_platforms` in [channel.hpp](https://github.com/mamba-org/mamba/blob/main/libmamba/include/mamba/core/channel.hpp#L23), it would make sense to rebind it to your changes (or replace its usage with your), to avoid duplication.

Reply:
Yes, I meant to use the code from this PR in the channel file; but this can be done in a dedicated PR.